docs/modules/ROOT/pages/documentation/payara-server/server-configuration/certificate-management.adoc

```diff
@@ -0,0 +1,490 @@
+|server
+|no
+
+|`--listener`
+|String
+|The name of the HTTP or IIOP listener to add the certificate to.
+|N/A
+|no
+
+|`--domain_name`
+|String
+|The name of the domain where the target instance exists.
+|domain1
+|no
+
+|`--domaindir`
+|String
+|The path to the directory containing the target domain.
+|${installDir}/glassfish/domains
+|no
+
+|`--file`
+|File
+|The certificate or bundle file to add to the target key store
+|N/A
+|yes
+
+|`--alias` (primary)
+|String
+|The alias name to store the certificate or bundle the key store under.
+|N/A
+|yes
+
+|===
+
+[[example]]
+=== Example
+
+[source, shell]
+----
+asadmin> add-to-keystore --file /home/anon/Downloads/mycert.p12 mycert
+----
+
+[[add-to-truststore]]
+== `add-to-truststore`
+
+*Usage*::
+`asadmin> add-to-truststore --target=instancename --listener=listenername --file /path/to/file alias`
+
+*Aim*::
+This command adds a certificate bundle (e.g. .p12 or .jks file) or certificate (e.g. .cert file) to the target instance
+or listener's trust store using the provided alias.
+
+If the instance or listener is configured to use the default trust store, the command will instead synchronise
+the instance with the DAS (under the assumption the certificate has been added to the default trust store of
+the DAS), since any certificates added to the instance stores would be lost upon next synchronisation.
+
+*Note*::
+This will overwrite an entry already present with the same alias.
+
+[[command-options]]
+=== Command Options
+
+[cols="3,1,5,1,1",options="header"]
+|===
+|Option
+|Type
+|Description
+|Default
+|Mandatory
+
+|`--target`
+|String
+|The name of the instance to add the certificate to.
+|server
+|no
+
+|`--listener`
+|String
+|The name of the HTTP or IIOP listener to add the certificate to.
+|N/A
+|no
+
+|`--domain_name`
+|String
+|The name of the domain where the target instance exists.
+|domain1
+|no
+
+|`--domaindir`
+|String
+|The path to the directory containing the target domain.
+|${installDir}/glassfish/domains
+|no
+
+|`--file`
+|File
+|The certificate or bundle file to add to the target trust store
+|N/A
+|yes
+
+|`--alias` (primary)
+|String
+|The alias name to store the certificate or bundle the trust store under.
+|N/A
+|yes
+
+|===
+
+[[example]]
+=== Example
+
+[source, shell]
+----
+asadmin> add-to-keystore --file /home/anon/Downloads/mycert.p12 mycert
+----
+
+[[remove-from-keystore]]
+== `remove-from-keystore`
+
+*Usage*::
+`asadmin> remove-from-keystore --target=instancename --listener=listenername alias`
+
+*Aim*::
+This command removes a certificate from the target instance or listener's key store matching the provided alias.
+
+If the instance or listener is configured to use the default key store, the command will instead synchronise
+the instance with the DAS (under the assumption the certificate has been removed from the default key store of
+the DAS), since any certificates removed from the instance stores would be lost upon next synchronisation.
```

Comment:
```suggestion
If the instance or listener is configured to use the default key store, the command will remove the certificate from the default key store of the DAS and then synchronise
the instance with the DAS. This is to preserve the certificate changes across instance restarts because any certificates removed directly from the instance stores would be added back upon next synchronisation.
```

Reply:
See comments elsewhere, your suggestion isn't what happens.